### PR TITLE
Implement host firewall for public vm interfaces

### LINKED ISSUE #1102: Proper host filtering and firewalling for vms with public IP's
follow up of #1098 
CC @delandtj for what actually needs to happen

### PR DESCRIPTION
**Pull request by LeeSmet**
Closes #1102

### CODE REVIEW

File: cmds/networkd/nft.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+)
+
+func ensureHostFw(ctx context.Context) error {
+	log.Info().Msg("ensuring existing host nft rules")
+	cmd := exec.CommandContext(ctx, "/bin/sh", "-c", "nft list ruleset")
```
Comment (muhamadazmy) [inline cmds/networkd/nft.go]:
```suggestion
	cmd := exec.CommandContext(ctx, "nft", "list", "ruleset")
```

File: pkg/provision/primitives/public_ip.go [hunk 2]
```diff
@@ -26,11 +34,122 @@ func (p *Provisioner) publicIPProvision(ctx context.Context, reservation *provis
...
+	}
 	return network.DisconnectPubTap(reservation.ID)
 }
+
+func filterName(reservationID string) string {
+	return fmt.Sprintf("r-%s", reservationID)
+}
+
+func setupFilters(ctx context.Context, fName string, iface string, ip string, ipv6 string, mac string) error {
+	cmd := exec.CommandContext(ctx, "/bin/sh", "-c",
```
Comment (muhamadazmy) [inline pkg/provision/primitives/public_ip.go:81]: Would it be nicer if we use go `text/template` to generate a template and write it directly to `nft` stdin, then just execute `nft -f -`

File: pkg/provision/primitives/public_ip.go [hunk 3]
```diff
@@ -26,11 +34,122 @@ func (p *Provisioner) publicIPProvision(ctx context.Context, reservation *provis
...
+nft 'add rule bridge filter %[1]s ip6 version 6 udp sport 547 drop'
+nft 'add rule bridge filter %[1]s ip version 4 udp sport 67 drop'`, fName, iface, ip, mac, ipv6))
+
+	if err := cmd.Run(); err != nil {
+		return errors.Wrap(err, "could not setup firewall rules for public ip")
+	}
+	return nil
+}
+
+func teardownFilters(ctx context.Context, fName string) error {
```
Comment (muhamadazmy) [inline pkg/provision/primitives/public_ip.go:107]: same comment as above

File: pkg/provision/primitives/public_ip.go (hunk 2 above)
Comment (LeeSmet) [reply pkg/provision/primitives/public_ip.go:81]: Not sure. Keep in mind that we are not just setting a template, but actually applying rules to an existing nft setup. I personally don't know nft enough to know if this is possible

**Review by muhamadazmy — APPROVED** [review]

**Pull request merged**